core/interop/pinvoke/marshalling/src/MarshallingSampleNative/MarshallingSampleNative.cpp

```diff
@@ -0,0 +1,356 @@
+{
+    unsigned char result = inValue + *inRef + *inOutRef;
+    *inOutRef = result;
+    *outRef = result;
+    return result;
+}
+
+extern "C" DLL_EXPORT char STDMETHODCALLTYPE SumSBytes(char inValue, /*[In]*/ char* inRef, /*[In,Out]*/ char* inOutRef, /*[Out]*/ char* outRef)
+{
+    char result = inValue + *inRef + *inOutRef;
+    *inOutRef = result;
+    *outRef = result;
+    return result;
+}
+
+extern "C" DLL_EXPORT unsigned short STDMETHODCALLTYPE SumUShorts(unsigned short inValue, /*[In]*/ unsigned short* inRef, /*[In,Out]*/ unsigned short* inOutRef, /*[Out]*/ unsigned short* outRef)
+{
+    unsigned short result = inValue + *inRef + *inOutRef;
+    *inOutRef = result;
+    *outRef = result;
+    return result;
+}
+
+extern "C" DLL_EXPORT short STDMETHODCALLTYPE SumShorts(short inValue, /*[In]*/ short* inRef, /*[In,Out]*/ short* inOutRef, /*[Out]*/ short* outRef)
+{
+    short result = inValue + *inRef + *inOutRef;
+    *inOutRef = result;
+    *outRef = result;
+    return result;
+}
+
+extern "C" DLL_EXPORT unsigned int STDMETHODCALLTYPE SumUInts(unsigned int inValue, /*[In]*/ unsigned int* inRef, /*[In,Out]*/ unsigned int* inOutRef, /*[Out]*/ unsigned int* outRef)
+{
+    unsigned int result = inValue + *inRef + *inOutRef;
+    *inOutRef = result;
+    *outRef = result;
+    return result;
+}
+
+extern "C" DLL_EXPORT int STDMETHODCALLTYPE SumInts(int inValue, /*[In]*/ int* inRef, /*[In,Out]*/ int* inOutRef, /*[Out]*/ int* outRef)
+{
+    int result = inValue + *inRef + *inOutRef;
+    *inOutRef = result;
+    *outRef = result;
+    return result;
+}
+
+extern "C" DLL_EXPORT uint64_t STDMETHODCALLTYPE SumULongs(uint64_t inValue, /*[In]*/ uint64_t* inRef, /*[In,Out]*/ uint64_t* inOutRef, /*[Out]*/ uint64_t* outRef)
+{
+    uint64_t result = inValue + *inRef + *inOutRef;
+    *inOutRef = result;
+    *outRef = result;
+    return result;
+}
+
+extern "C" DLL_EXPORT int64_t STDMETHODCALLTYPE SumLongs(int64_t inValue, /*[In]*/ int64_t* inRef, /*[In,Out]*/ int64_t* inOutRef, /*[Out]*/ int64_t* outRef)
+{
+    int64_t result = inValue + *inRef + *inOutRef;
+    *inOutRef = result;
+    *outRef = result;
+    return result;
+}
+
+extern "C" DLL_EXPORT float STDMETHODCALLTYPE SumFloats(float inValue, /*[In]*/ float* inRef, /*[In,Out]*/ float* inOutRef, /*[Out]*/ float* outRef)
+{
+    float result = inValue + *inRef + *inOutRef;
+    *inOutRef = result;
+    *outRef = result;
+    return result;
+}
+
+extern "C" DLL_EXPORT double STDMETHODCALLTYPE SumDoubles(double inValue, /*[In]*/ double* inRef, /*[In,Out]*/ double* inOutRef, /*[Out]*/ double* outRef)
+{
+    double result = inValue + *inRef + *inOutRef;
+    *inOutRef = result;
+    *outRef = result;
+    return result;
+}
+
+
+#if !WINDOWS
+typedef struct tagDEC {
+    // Decimal.cs treats the first two shorts as one long
```

Comment:
Maybe reword this comment as "the runtime" instead of referencing a specific file in CoreCLR?

Reply:
I deleted the comment - this is sort of a niche thing anyway.